utilities.h

```diff
@@ -22,6 +22,20 @@ class Utilities
 {
 public:
 
+    static constexpr float constexpr_sqrt(float x, float epsilon = 1e-5f)
```

Comment:
I was under the impression that std::sqrt is constexpr in C++20 but couldn't get it to work for the life of me, so included this version.  It would be preferable to use the STL version if it can be done...

Reply:
Just responding to this one for now: the sqrt functions are actually constexpr starting from c++_26_...